daft/expressions/expressions.py

```diff
@@ -2922,6 +2922,17 @@ def join(self, delimiter: str | Expression) -> Expression:
         delimiter_expr = Expression._to_expression(delimiter)
         return Expression._from_pyexpr(native.list_join(self._expr, delimiter_expr._expr))
 
+    def value_counts(self) -> Expression:
+        """Counts the occurrences of each unique value in the list.
+
+        Returns:
+            Expression: A list of structs, where each struct contains a 'value' field
+                        representing a unique element from the original list, and a 'count' field
+                        representing the number of times that value appears in the list.
```

Comment:
It should be an Expression with a `Map<X, Int32>` type, where `List<X>` is the input type!

Reply:
78c47eb72effbc7804cb825dc3460adb7a637b22